### ISSUE TITLE
Possibility to translate "message" properties of json configs

### ISSUE THREAD

**Issue by ghost** [stylelint/stylelint#7472]
### What is the problem you're trying to solve?

When a custom stylelint config is saved in .js format, we are able to write the message property in 2 ways:

### Scenario 1: JS - Hardcoded message with variable
```
export const customConfig = {
  'unit-disallowed-list': [
    [
      'px',
    ],
    {
      message: 'print this custom message when this error is triggered: { unit }',
    },
  ],
};
```
  
  or even:
  
### Scenario 2: JS - translated message with variable
```
import i18next from 'i18next';
import translationFile from './pt-BR.json';

i18next.init({
  debug: false,
  lng: 'es',
  resources: {
    es: {
      translation: translationFile,
    },
  },
});
  
export const customConfig = {
  'unit-disallowed-list': [
    [
      'px',
    ],
    {
      message: (unit) => {
        return i18next.t('unit-disallowed-list', { unit });
      },
    },
  ],
};
```

Both scenarios are going to be applied as expected:
```
import customConfig from './customConfig.js';

export const stylelintConfig = {
  rules: {
    ...customConfig,
  }
};
```

And in the terminal, both the variables and translated messages are processed when they are printed.

------

My doubt is: instead of saving customConfig in .js format, when saving it in .json format, how can I continue to achieve these 2 scenarios?

### Attempt 1: Json - Hardcoded message with variable
```
...
"message": "print this custom message when this error is triggered: { unit }",
...
```

The variable {unit} is no longer recognized by Stylelint.

### Attemp 2: Json - translated message with variable
```
...
"message": "(unit) => { return i18next.t('unit-disallowed-list', { unit }); }",
...
```

or

```
...
"message": "{{translatedMessage}} { unit }",
...
```

The variable {unit} is no longer recognized by Stylelint, so I can't even try to translate it with an external library like i18next.

### What solution would you like to see?

A solution I would like to achieve for the short term is to be able to: convert a custom Stylelint config from the .js format to the .json format, and to maintain the possibility of saving a custom message that contains default variables provided by Stylelint, such as {unit} in the example above.
Better yet, would be if: 1 - Stylelint could provide translations by default, so I don't need to deal with these conversions of file format and inserting a variable in the .json file;    .. and if Stylelint doesn't want to include translations by default, then it would still be good if Stylelint could at least provide a helper function to enable us to provide a translated .js or .json file fullfilled with messages in another language, instead of english, and this somehow would be merged with custom configs that exist in the .json format. As a result, when it comes the time to print messages in terminal, these would be shown translated.

**Comment by ybiquitous** [comment]
In JSON config, can you try `%s`? Document:
https://github.com/stylelint/stylelint/blob/38e0606d0e9d3e51e9e37f5cd87abca4b8330875/docs/user-guide/configure.md?plain=1#L162-L173

**Comment by RaniaLey** [comment]
This more immediate request worked for now. Thank you. I didn't find this information before...

I believe it remain open other suggestions in this thread.

**Comment by RaniaLey** [comment]
Important to say that the variable  \"%s\", as provided by @ybiquitous, doesn't work currently in all rules. This seem to be a Stylelint bug. For example: "color-named" rule. A translated message for this rule, saved in .json file, is correctly being shown in the terminal, except its  \"%s\" variable.

**Comment by ybiquitous** [comment]
Right, the feature is still experimental, so there are some rules not supporting it. Since we have a plan to stabilize this feature (#6966), we welcome your contribution if you're interested.

**Comment by jeddy3** [comment]
Closing as a dupe of https://github.com/stylelint/stylelint/issues/6966.

Please consider [contributing](https://stylelint.io/contributing) to that issue if you have time so that we can provide the feature for all rules.
